internal/template-validator/webhooks/utils.go

```diff
@@ -69,29 +70,35 @@ func ToAdmissionResponse(causes []metav1.StatusCause) *admissionv1.AdmissionResp
 	}
 }
 
-func GetAdmissionReviewVM(ar *admissionv1.AdmissionReview) (*kubevirt.VirtualMachine, *kubevirt.VirtualMachine, error) {
+func GetAdmissionReviewVM(ar *admissionv1.AdmissionReview) (*kubevirt.VirtualMachine, error) {
 	if ar.Request.Resource.Resource != "virtualmachines" {
-		return nil, nil, fmt.Errorf("expect resource %v to be '%s'", ar.Request.Resource, "virtualmachines")
+		return nil, fmt.Errorf("expect resource %v to be '%s'", ar.Request.Resource, "virtualmachines")
 	}
 
-	var err error
-	raw := ar.Request.Object.Raw
-	newVM := kubevirt.VirtualMachine{}
-
-	err = json.Unmarshal(raw, &newVM)
+	newVM := &kubevirt.VirtualMachine{}
+	err := json.Unmarshal(ar.Request.Object.Raw, newVM)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	if ar.Request.Operation == admissionv1.Update {
-		raw := ar.Request.OldObject.Raw
-		oldVM := kubevirt.VirtualMachine{}
-		err = json.Unmarshal(raw, &oldVM)
-		if err != nil {
-			return nil, nil, err
-		}
-		return &newVM, &oldVM, nil
+	return newVM, nil
+}
+
+func GetAdmissionReviewTemplate(ar *admissionv1.AdmissionReview) (*templatev1.Template, error) {
+	const resourceName = "templates"
+	if ar.Request.Resource.Resource != resourceName {
+		return nil, fmt.Errorf("expected resource %v to be '%s'", ar.Request.Resource, resourceName)
 	}
 
-	return &newVM, nil, nil
+	obj := &ar.Request.Object
+	if ar.Request.Operation == admissionv1.Delete {
+		obj = &ar.Request.OldObject
+	}
+
+	template := &templatev1.Template{}
+	err := json.Unmarshal(obj.Raw, template)
+	if err != nil {
+		return nil, err
+	}
+	return template, nil
```

Comment:
you can use directly
`return template, err`

Reply:
Again, `template` is not `nil` in case of error.

Reply:
Yeah, I'd only return a variable if it's content is supposed to be of value in the return.
So do things like `return template, err` if the template value still might be used afterwards.
Only return one of them if only one value is supposed to be used.
Doesn't matter if the value should be nil anyways, it's just a nice convention to make sure the consumer never gets a value they should not use and the next developer knows that it is intentional.